data/transactions/logic/eval.go

```diff
@@ -2611,27 +2639,28 @@ func (cx *EvalContext) getLatestTimestamp() (timestamp uint64, err error) {
 	return uint64(ts), nil
 }
 
-func (cx *EvalContext) getApplicationAddress() basics.Address {
-	// Initialize appAddrCache if necessary
-	if cx.appAddrCache == nil {
-		cx.appAddrCache = make(map[basics.AppIndex]basics.Address)
-	}
-
-	appAddr, ok := cx.appAddrCache[cx.appID]
+// getApplicationAddress memoizes app.Address() across a tx group's evaluation
+func (cx *EvalContext) getApplicationAddress(app basics.AppIndex) basics.Address {
```

Comment:
Widened this method to cache app addresses for any given app id, so that the check for use of a created app id will be fast when repeated.